indico/modules/events/registration/templates/management/_registration_details.html

```diff
@@ -169,7 +169,8 @@
     {% set can_reset = registration.registration_form.moderation_enabled and not registration.is_paid %}
     {% set ticket_template = registration.registration_form.get_ticket_template() %}
     {% set can_get_ticket = registration.registration_form.tickets_enabled and (not ticket_template.is_ticket or not registration.is_ticket_blocked) %}
-    {% if can_reset or can_get_ticket %}
+    {% set can_get_document = registration.registration_form.has_templates() %}
```

Comment:
I just noticed that you put this in `_render_complete_actions` - but it's perfectly valid to generate a document for a registration regardless of its stage.

Reply:
yeah makes sense, 
after seeing the issue, I though it was to be always with `get ticket`.

Should i separate this button out in a macro and call the macro in all toolbars(regardless of reg stage)?

Reply:
How about showing it in the "documents" box instead?

Reply:
makes sense.
This means changing the behaviour of the documents box as well, as it's currently only rendered when docs are available.

Reply:
Yes, I'd render it when either documents or document templates are available.